tests/test_core.py

```diff
@@ -1543,6 +1543,63 @@ def test_exceptions_rethrow_missing(self):
     create_file('main.cpp', 'int main() { throw; }')
     self.do_runf('main.cpp', None, assert_returncode=NON_ZERO)
 
+  def test_format_exception(self):
+    # needs to flush stdio streams
+    self.set_setting('EXIT_RUNTIME')
+    self.set_setting('EXPORTED_FUNCTIONS', ['_main', '_throw_exc'])
+    self.set_setting('DISABLE_EXCEPTION_CATCHING', 0)
+    self.set_setting('DEFAULT_LIBRARY_FUNCS_TO_INCLUDE', ['$formatException'])
+    self.maybe_closure()
+    src = '''
+      #include <emscripten.h>
+      #include <exception>
+      #include <stdexcept>
+      using namespace std;
+
+      class myexception : public exception {
+        virtual const char* what() const throw() { return "My exception happened"; }
+      } myex;
+
+      extern "C" void throw_exc(int x) {
```

Comment:
I think you can add `EMSCRIPTEN_KEEKALIVE` here and then from the `EXPORTED_FUNCTIONS` line above.